Add agent version filter option (#6312)
wazuh/wazuh-documentation: source/user-manual/api/configuration.rst

.. versionadded:: 4.6.0

+------------+----------------------+---------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Sub-fields | Allowed values       | Default value | Description                                                                                                                                                                                                                                                                                                                                 |
+============+======================+===============+=============================================================================================================================================================================================================================================================================================================================================+

> | allow      | yes, true, no, false | true          | Allow uploading configurations accepting higher agent versions through the Wazuh API. Setting this option to ``false`` prevents uploading ``ossec.conf`` files that contain the ``<agents><allow_higher_versions>`` section with the ``yes`` value inside the :ref:`global tag <reference_ossec_global>`.                                   |

Reviewer: ```suggestion
| allow      | yes, true, no, false | true          | Allow uploading configurations that accept higher agent versions through the Wazuh API. Setting this option to ``false`` prevents uploading ``ossec.conf`` files that contain the ``<agents><allow_higher_versions>`` section with the ``yes`` value inside the :ref:`global tag <reference_ossec_global>`.                                 |
```

> | allow      | yes, true, no, false | true          | Allow uploading configurations that accept higher agent versions through the Wazuh API. Setting this option to ``false`` prevents uploading ``ossec.conf`` files that contain the ``<agents><allow_higher_versions>`` section with the ``yes`` value inside the :ref:`global tag <reference_ossec_global>`.                                 |

Reviewer 2: After the last changes made [here](https://github.com/wazuh/wazuh/pull/18342), this part needs to be updated:

```suggestion
| allow      | yes, true, no, false | true          | Allow uploading configurations that accept higher agent versions through the Wazuh API. Setting this option to ``false`` prevents uploading ``ossec.conf`` files that contain the ``<allow_higher_versions>`` section with the ``yes`` value inside the :ref:`auth <reference_ossec_auth>` or :ref:`remote <reference_ossec_remote>` tags.  |
```

> | allow      | yes, no              | yes          | Allow uploading configurations accepting higher agent versions through the Wazuh API. Setting this option to ``false`` prevents uploading ``ossec.conf`` files that contain the ``<agents><allow_higher_versions>`` section inside the :ref:`global tag <reference_ossec_global>`.                                                                                 |

Reviewer 2: ```suggestion
| allow      | yes, true, no, false | true          | Allow uploading configurations accepting higher agent versions through the Wazuh API. Setting this option to ``false`` prevents uploading ``ossec.conf`` files that contain the ``<agents><allow_higher_versions>`` section with value ``yes`` inside the :ref:`global tag <reference_ossec_global>`.                                       |
```
